### PR TITLE
#673 Enhance mobile device experience

### LINKED ISSUE #673: 建议优化下移动端（手机,平板）地铁线路绘制
https://github.com/railmapgen/rmp/assets/162086921/b3eea62c-2a6c-4ddf-ab66-07cc39801ab9

移动端制作线路图划屏幕滑不过来，点旁边的选项还会报错，建议优化一下

### PR DESCRIPTION
**Pull request by langonginc**
Fix #673

### CODE REVIEW

File: src/components/svg-canvas-graph.tsx [hunk 1]
```diff
@@ -60,40 +60,21 @@ const SvgCanvas = () => {
...
-            }
-        }
         // console.log('down ', graph.current.getNodeAttributes(node));
     });
     const handlePointerMove = useEvent((node: StnId | MiscNodeId, e: React.PointerEvent<SVGElement>) => {
         const { x, y } = getMousePosition(e);
 
         if (mode === 'free' && active === node) {
-            selected.forEach(s => {
-                if (graph.current.hasNode(s)) {
```
Comment (thekingofcity) [inline src/components/svg-canvas-graph.tsx:89]: Can't we just `if (graph.current.hasNode(s) && s === active) {` ?

Comment (langonginc) [reply src/components/svg-canvas-graph.tsx:89]: The node isn't in selected while moving.

I put the "adding node to selected" into "onPointerUp", as the right sidebar will appear and cover the canvas on mobile devices.

Comment (thekingofcity) [reply src/components/svg-canvas-graph.tsx:89]: Fare enough. Let's simplify this to `[selected.has(active) ? undefined : active, ...selected]`.

Comment (langonginc) [reply src/components/svg-canvas-graph.tsx:89]: I suddenly think it should be `(selected.has(active) ? selected : [active])`.

- move one node => no selected, do only for `active`.
- move many node => nodes are in `selected`, loop in `selected` .

Comment (thekingofcity) [reply src/components/svg-canvas-graph.tsx:89]: If you only move `[active]`, other selected in the multiple selections will not be moved, right?

Comment (langonginc) [reply src/components/svg-canvas-graph.tsx:89]: They will all be moved, because `selected` has `active` .

Comment (thekingofcity) [reply src/components/svg-canvas-graph.tsx:89]: Then how about when I already have multiple selections on stn1 and stn2 and then move stn3 with the shift key pressed? In the main branch, the stn3 should be moved as well.

Comment (langonginc) [reply src/components/svg-canvas-graph.tsx:89]: I'll rollback these changes, and just hide the panel when active is not `undefined` to make sure no panel covering the canvas while moving nodes.

File: src/components/svg-canvas-graph.tsx [hunk 2]
```diff
@@ -84,8 +84,12 @@ const SvgCanvas = () => {
     const handlePointerMove = useEvent((node: StnId | MiscNodeId, e: React.PointerEvent<SVGElement>) => {
         const { x, y } = getMousePosition(e);
 
+        if (active === node && !selected.has(active)) {
+            dispatch(setSelected(new Set()));
+        }
+
```
Comment (thekingofcity) [inline src/components/svg-canvas-graph.tsx:90]: Also here.

**Review by thekingofcity — CHANGES_REQUESTED** [review]
Let's add some comments

**Review by thekingofcity — APPROVED** [review]
LGTM

**Pull request merged**